### ISSUE TITLE
Implement device parameter in Dropout2d

### ISSUE THREAD

**Issue by Bengt** [pytorch/pytorch#111368]
### 🚀 The feature, motivation and pitch

Please provide a `device` parameter to the Dropout2d layer for creating it on a target device like so:

```python
dropout = nn.Dropout2d(
    p=0.5,
    device=torch.device('cuda'),
)
```

### Alternatives

Currently, [PyTorch's Dropout2D layer](https://pytorch.org/docs/stable/generated/torch.nn.Dropout2d.html) does not support creating it on a specific device. This means that one needs first create it on the CPU and then move it to the desired device, if any. Lacking this feature results in extra code to be written by the user:

```python
dropout = nn.Dropout2d(
    p=0.5,
)
dropout1.cuda()
```

### Additional context

Creating the layer on CPU and moving it to GPU might also be slower, when compared to creation directly on the target device, as hinted at by the performance tuning guide:

<https://pytorch.org/tutorials/recipes/recipes/tuning_guide.html#create-tensors-directly-on-the-target-device>

**Comment by SiddharthIVEX** [comment]
I can take it.

**Comment by Siddharth11235** [comment]
I can take it.

**Comment by Siddharth11235** [comment]
Quick question: I finally had time to add the feature, but how do I verify that the dropout fn is created on the GPU?
